specs/opcode/12SLT_13SGT.md

```diff
@@ -0,0 +1,62 @@
+# SLT & SGT opcode
+
+## Procedure
+
+The `SLT` and `SGT` opcodes compare the top two values on the stack, and push the result (0 or 1) back to the stack.
+
+The stack inputs `a` and `b` are 256-bits signed integers with the most significant bit being the sign (1 for negative and 0 for positive). The 256-bits are in the two's complement form of representing signed integers.
+
+#### Circuit Behaviour
+
+The `SignedComparatorGadget` takes arguments `a: [u8; 32]`, `b: [u8; 32]` and `is_sgt: bool`.
+
+It returns the result of `a < b` where:
+
+- `Stack = [b, a]` if `is_sgt == false`
+- `Stack = [a, b]` if `is_sgt == true`
+
+The gadget is constructed with the following logic:
+
+```
+if a_0 >= 128 and b_0 < 128:
```

Comment:
It seems the comparisons are swapped in `src/zkevm_specs/evm/execution/slt_sgt.py`

Here it's 
```
if a_0 >= 128 and b_0 < 128:
```
In `slt_sgt.py` it's:
```
if a8s[31] < 128 and b8s[31] >= 128:
```

And the same happens for the elif.

Reply:
Missed this.. My bad.

Reply:
Depending on whether `is_sgt` is true or false, the comparisons change, but I think it's better to be consistent in both places

Reply:
The problem is that circuit uses little-endian encoding while spec uses big-endian. We could bring this as a separate issue to make it consistent between circuit and spec.

Reply:
> The problem is that circuit uses little-endian encoding while spec uses big-endian. We could bring this as a separate issue to make it consistent between circuit and spec.

Hm, in the issue I raised I'm not worried about the endianness, but about swapping `>=` with `<`, and `< with `>=`, which change the meaning from SGT to SLT.

Reply:
@ed255 The mismatch between spec and circuit was my bad. I have updated the specs, as well as the Python implementation of the circuit to now swap `a` and `b` depending on whether or not the opcode is `SGT` (just like we implement in the [`zkevm-circuits` PR](https://github.com/appliedzkp/zkevm-circuits/pull/280)).

Also, the most-significant byte will now be `a[31]` (justl like the Rust impl), because after rebasing this PR, the `rlc_to_bytes` has been replaced by `rlc_to_le_bytes`.

Also, I've made use of `instruction.compare` wherever required, removing the dynamically generated constraints.